src/lib.rs

```diff
@@ -99,9 +97,276 @@ use std::fmt;
 use std::iter::repeat;
 use std::result;
 
+/// A description of the options that a program can handle
+pub struct Options {
+    grps: Vec<OptGroup>
+}
+
+impl Options {
+    /// Create a blank set of options
+    pub fn new() -> Options {
+        Options {
+            grps: Vec::new()
+        }
+    }
+
+    /// Create a generic option group, stating all parameters explicitly
+    pub fn add_opt<'a>(&'a mut self, short_name: &str, long_name: &str, desc: &str,
```

Comment:
Most builder setters are simply just "foo" instead of something like "set_foo" or "add_foo"